### ISSUE TITLE
Question about generating validation dataset.

### ISSUE THREAD

**Issue by SingCheng** [wenyyu/Image-Adaptive-YOLO#70]
According to step two which is used to generate ten levels' foggy training images and val images, respectively

However, when we access data_make.py. We can observe that the img_name for the validation set has been commented.
![1683810902421](https://github.com/wenyyu/Image-Adaptive-YOLO/assets/50645313/7af6161e-7d0f-46af-b7ce-2fe1053dfec7)

The VOC structure is exactly the same as it is required.

is there anything I have missed?

**Comment by lyq216398** [comment]
> According to step two which is used to generate ten levels' foggy training images and val images, respectively
> 
> However, when we access data_make.py. We can observe that the img_name for the validation set has been commented. ![1683810902421](https://private-user-images.githubusercontent.com/50645313/237699819-7af6161e-7d0f-46af-b7ce-2fe1053dfec7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExNzI4NzksIm5iZiI6MTcwMTE3MjU3OSwicGF0aCI6Ii81MDY0NTMxMy8yMzc2OTk4MTktN2FmNjE2MWUtN2QwZi00NmFmLWI3Y2UtMmZlMTA1M2RmZWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI4VDExNTYxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiNzE1NmE1ZmZjNTliMTcxNTg3ZWM3MWUxODVmMGQxMDc5M2M0ZWI4OGIyMjdlNDJkYWVlZjRmZmU0ZDA0ZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rTsJXs6bTiQdybtd4X7cf138KIARFHw06CQSA1k-6AM)
> 
> The VOC structure is exactly the same as it is required.
> 
> is there anything I have missed?

I have the same question.Did you solve it?

**Comment by SingCheng** [comment]
> > According to step two which is used to generate ten levels' foggy training images and val images, respectively
> > However, when we access data_make.py. We can observe that the img_name for the validation set has been commented. ![1683810902421](https://private-user-images.githubusercontent.com/50645313/237699819-7af6161e-7d0f-46af-b7ce-2fe1053dfec7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExNzI4NzksIm5iZiI6MTcwMTE3MjU3OSwicGF0aCI6Ii81MDY0NTMxMy8yMzc2OTk4MTktN2FmNjE2MWUtN2QwZi00NmFmLWI3Y2UtMmZlMTA1M2RmZWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI4VDExNTYxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiNzE1NmE1ZmZjNTliMTcxNTg3ZWM3MWUxODVmMGQxMDc5M2M0ZWI4OGIyMjdlNDJkYWVlZjRmZmU0ZDA0ZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rTsJXs6bTiQdybtd4X7cf138KIARFHw06CQSA1k-6AM)
> > The VOC structure is exactly the same as it is required.
> > is there anything I have missed?
> 
> I have the same question.Did you solve it?

Hi @lyq216398 i did solved it, if you are generating validation set, just uncomment it and comment the the line above.

**Comment by lyq216398** [comment]
> > > According to step two which is used to generate ten levels' foggy training images and val images, respectively
> > > However, when we access data_make.py. We can observe that the img_name for the validation set has been commented. ![1683810902421](https://private-user-images.githubusercontent.com/50645313/237699819-7af6161e-7d0f-46af-b7ce-2fe1053dfec7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExNzI4NzksIm5iZiI6MTcwMTE3MjU3OSwicGF0aCI6Ii81MDY0NTMxMy8yMzc2OTk4MTktN2FmNjE2MWUtN2QwZi00NmFmLWI3Y2UtMmZlMTA1M2RmZWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI4VDExNTYxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiNzE1NmE1ZmZjNTliMTcxNTg3ZWM3MWUxODVmMGQxMDc5M2M0ZWI4OGIyMjdlNDJkYWVlZjRmZmU0ZDA0ZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rTsJXs6bTiQdybtd4X7cf138KIARFHw06CQSA1k-6AM)
> > > The VOC structure is exactly the same as it is required.
> > > is there anything I have missed?
> > 
> > 
> > I have the same question.Did you solve it?
> 
> Hi @lyq216398 i did solved it, if you are generating validation set, just uncomment it and comment the the line above.

Thank you for your reply!I want to know whether the validation set is for final testing or just is separated from the training set to verify the effectiveness during training.I am a little confused with the concept of "val and test".

**Comment by SingCheng** [comment]
> > > > According to step two which is used to generate ten levels' foggy training images and val images, respectively
> > > > However, when we access data_make.py. We can observe that the img_name for the validation set has been commented. ![1683810902421](https://private-user-images.githubusercontent.com/50645313/237699819-7af6161e-7d0f-46af-b7ce-2fe1053dfec7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExNzI4NzksIm5iZiI6MTcwMTE3MjU3OSwicGF0aCI6Ii81MDY0NTMxMy8yMzc2OTk4MTktN2FmNjE2MWUtN2QwZi00NmFmLWI3Y2UtMmZlMTA1M2RmZWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI4VDExNTYxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiNzE1NmE1ZmZjNTliMTcxNTg3ZWM3MWUxODVmMGQxMDc5M2M0ZWI4OGIyMjdlNDJkYWVlZjRmZmU0ZDA0ZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rTsJXs6bTiQdybtd4X7cf138KIARFHw06CQSA1k-6AM)
> > > > The VOC structure is exactly the same as it is required.
> > > > is there anything I have missed?
> > > 
> > > 
> > > I have the same question.Did you solve it?
> > 
> > 
> > Hi @lyq216398 i did solved it, if you are generating validation set, just uncomment it and comment the the line above.
> 
> Thank you for your reply!I want to know whether the validation set is for final testing or just is separated from the training set to verify the effectiveness during training.I am a little confused with the concept of "val and test".

The validation set is to verify the effectiveness during training. The author has mixing up the val and test in his code, if i'm remember correctly.

**Comment by lyq216398** [comment]
> > > > > According to step two which is used to generate ten levels' foggy training images and val images, respectively
> > > > > However, when we access data_make.py. We can observe that the img_name for the validation set has been commented. ![1683810902421](https://private-user-images.githubusercontent.com/50645313/237699819-7af6161e-7d0f-46af-b7ce-2fe1053dfec7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExNzI4NzksIm5iZiI6MTcwMTE3MjU3OSwicGF0aCI6Ii81MDY0NTMxMy8yMzc2OTk4MTktN2FmNjE2MWUtN2QwZi00NmFmLWI3Y2UtMmZlMTA1M2RmZWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTI4VDExNTYxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiNzE1NmE1ZmZjNTliMTcxNTg3ZWM3MWUxODVmMGQxMDc5M2M0ZWI4OGIyMjdlNDJkYWVlZjRmZmU0ZDA0ZTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rTsJXs6bTiQdybtd4X7cf138KIARFHw06CQSA1k-6AM)
> > > > > The VOC structure is exactly the same as it is required.
> > > > > is there anything I have missed?
> > > > 
> > > > 
> > > > I have the same question.Did you solve it?
> > > 
> > > 
> > > Hi @lyq216398 i did solved it, if you are generating validation set, just uncomment it and comment the the line above.
> > 
> > 
> > Thank you for your reply!I want to know whether the validation set is for final testing or just is separated from the training set to verify the effectiveness during training.I am a little confused with the concept of "val and test".
> 
> The validation set is to verify the effectiveness during training. The author has mixing up the val and test in his code, if i'm remember correctly.

Thank you !!!
